### PR TITLE
Explicitly support python 3.9->3.12

### PR DESCRIPTION
**Pull request by jhkennedy**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -20,6 +20,8 @@ classifiers=[
     "Programming Language :: Python :: 3.8",
```
Comment (jtherrmann) [inline pyproject.toml:20]:
```suggestion
```

**Review by jtherrmann — DISMISSED** [review]

**Review by jtherrmann — CHANGES_REQUESTED** [review]
We'll also need to update `environment.yml`. Also, does the `version` field in the Jupyter notebooks under `docs/` need to be updated? (I just ran a `git grep '3\.8'`.)

**Comment by jhkennedy** [comment]
@jtherrmann I updated `environment.yml`.  The `version` field in the notebooks is just metadata recorded in the notebook about the Python version when it was run and checked into the repo. I don't see a need to update it, and it's not something you can every really see. It is possible to strip, albeit annoying to do so.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [PEP 440](https://www.python.org/dev/peps/pep-0440/) 
 and uses [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+
+## [3.2.0]
+### Added
+* The HyP3 SDK now explicitly supports python 3.8-3.12
```
Comment (jhkennedy) [inline CHANGELOG.md:12]:
```suggestion
* The HyP3 SDK now explicitly supports python 3.9-3.12
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [PEP 440](https://www.python.org/dev/peps/pep-0440/) 
 and uses [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+
+## [3.2.0]
+### Added
+* The HyP3 SDK now explicitly supports python 3.9-3.12
```
Comment (jhkennedy) [inline CHANGELOG.md:12]:
```suggestion
* The HyP3 SDK now explicitly supports Python 3.9-3.12
```

**Review by jtherrmann — APPROVED** [review]

**Pull request merged**